enhancements/dns/nodeservice-etc-hosts-injection.md

```diff
@@ -0,0 +1,422 @@
+   - Uses atomic file operations to prevent corruption
+6. When the `nodeServices` list changes in the DNS CR, the operator updates the DaemonSet `SERVICES` environment variable, triggering a rolling restart to pick up the new service list
+
+Steps 1-4 are the only steps modified by this enhancement.
+
+#### Validation
+
+The following validation rules will be enforced:
+
+- `namespace` and `name` fields must not be empty
+- Service names must conform to DNS label standards (RFC 1123)
+- Namespace names must conform to Kubernetes namespace naming requirements
+- Maximum of 20 additional services can be specified
+  - This limit is imposed due to the implementation storing service names in the DaemonSet `SERVICES` environment variable
+  - Kubernetes has practical limits on environment variable sizes (typically 128KB total for all variables)
+  - Each service entry requires storage for the service name in `service-name.namespace.svc` format only (IPs are looked up dynamically)
+  - Example: `image-registry.openshift-image-registry.svc` requires approximately 30-70 bytes
+  - 20 services × ~70 bytes = ~1.4KB, leaving ample headroom for other environment variables
+  - The limit provides a safety margin and aligns with the intended use case (critical infrastructure services)
+
+#### Backward Compatibility
+
+If the `nodeServices` field is not specified or is empty, the DNS operator will use a default list containing the cluster image registry service to maintain backward compatibility with existing clusters. This ensures that existing functionality is preserved during upgrades.
+
+#### File Management
+
+The DNS node resolver will:
+- Use atomic file operations (write to temp file, then copy) to prevent corruption
+- Preserve the original file's attributes (permissions, ownership)
+- Add a comment marker (`# openshift-generated-node-resolver`) to identify managed entries
+- Filter out old managed entries before adding updated ones
+- Only update `/etc/hosts` if valid service IPs are resolved
+
+#### Environment Variable Considerations
+
+The implementation uses a Kubernetes DaemonSet environment variable named `SERVICES` to pass a list of service hostnames to the DNS node resolver pods. The `update-node-resolver.sh` script then dynamically looks up the IP addresses for these services. This approach has several important implications:
+
+**Advantages:**
+- Simple and reliable mechanism built into Kubernetes
+- No need for shared storage or complex synchronization
+- Standard Kubernetes primitives for configuration management
+- **Service IP changes don't require pod restarts**: The script detects and updates IPs dynamically via 60-second polling
+- Small footprint: only service names in `service-name.namespace.svc` format are stored, not IP addresses
+- Efficient: ~1.4KB for 20 services, minimal impact on environment variable limits
+- Flexible format: supports comma-separated, space-separated, or mixed delimiters
```

Comment:
I don't think this is actually a benefit. Ambiguity in an API and having multiple ways to define the same things is often confusing